File: gaming/src/main/java/com/oruga/gaming/dto/MensajeDto.java
package com.oruga.gaming.dto;

import com.oruga.gaming.entity.TipoMensaje;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class MensajeDto {
    private Long id;
    private Long remitenteId;
    private String remitenteNombre;
    private Long destinatarioId;
    private String contenido;
    private String imagenUrl;
    private LocalDateTime fechaEnvio;
    private Boolean esGrupal;
    private Boolean leido;
    private TipoMensaje tipoMensaje;
    private String tiempoRelativo;
}

File: MyApplication3/app/src/main/java/com/example/myapplication/data/remote/dto/MensajeDTO.kt
package com.example.myapplication.data.remote.dto

data class EnviarMensajeGrupalRequest(
    val contenido: String,
    val imagenUrl: String = ""
)

data class EnviarMensajePrivadoRequest(
    val destinatarioId: Long,
    val contenido: String,
    val imagenUrl: String = ""
)

data class MensajeResponse(
    val id: Long,
    val remitenteId: Long,
    val remitenteNombre: String,
    val contenido: String,
    val imagenUrl: String,
    val fechaEnvio: String,
    val esGrupal: Boolean,
    val tipoMensaje: String
)
